internal/services/storagemover/storage_mover_source_endpoint_resource.go

```diff
@@ -0,0 +1,257 @@
+	return "azurerm_storage_mover_source_endpoint"
+}
+
+func (r StorageMoverSourceEndpointResource) ModelObject() interface{} {
+	return &StorageMoverSourceEndpointModel{}
+}
+
+func (r StorageMoverSourceEndpointResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return endpoints.ValidateEndpointID
+}
+
+func (r StorageMoverSourceEndpointResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[0-9a-zA-Z][-_0-9a-zA-Z]{0,63}$`),
+				`The name must be between 1 and 64 characters in length, begin with a letter or number, and may contain letters, numbers, dashes and underscore.`,
+			),
+		},
+
+		"storage_mover_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: storagemovers.ValidateStorageMoverID,
+		},
+
+		"host": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"export": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"nfs_version": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			Default:  string(endpoints.NfsVersionNFSauto),
+			ValidateFunc: validation.StringInSlice([]string{
+				string(endpoints.NfsVersionNFSauto),
+				string(endpoints.NfsVersionNFSvFour),
+				string(endpoints.NfsVersionNFSvThree),
+			}, false),
+		},
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (r StorageMoverSourceEndpointResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r StorageMoverSourceEndpointResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model StorageMoverSourceEndpointModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.StorageMover.EndpointsClient
+			storageMoverId, err := storagemovers.ParseStorageMoverID(model.StorageMoverId)
+			if err != nil {
+				return err
+			}
+
+			id := endpoints.NewEndpointID(storageMoverId.SubscriptionId, storageMoverId.ResourceGroupName, storageMoverId.StorageMoverName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := endpoints.Endpoint{
+				Name: utils.String(model.Name),
```

Comment:
per the ARM spec it's not valid to specify `name` in the payload

```suggestion
```

Reply:
Fixed.